libraries/core_libs/consensus/src/pbft/block_proposer.cpp

```diff
@@ -198,13 +198,26 @@ level_t BlockProposer::getProposeLevel(blk_hash_t const& pivot, vec_blk_t const&
 void BlockProposer::proposeBlock(DagFrontier&& frontier, level_t level, SharedTransactions&& trxs, VdfSortition&& vdf) {
   if (stopped_) return;
 
-  vec_trx_t trx_hashes;
-  std::transform(trxs.begin(), trxs.end(), std::back_inserter(trx_hashes),
-                 [](std::shared_ptr<Transaction> const& t) { return t->getHash(); });
+  auto proposal_period = db_->getProposalPeriodForDagLevel(level);
 
   // When we propose block we know it is valid, no need for block verification with queue,
   // simply add the block to the DAG
-  DagBlock blk(frontier.pivot, level, std::move(frontier.tips), std::move(trx_hashes), std::move(vdf), node_sk_);
+  vec_trx_t trx_hashes;
+  estimations_vec_t estimations;
+  u256 block_weight = 0;
+  for (auto trx : trxs) {
+    const auto& trx_hash = trx->getHash();
+    auto weight = trx_mgr_->estimateTransactionByHash(trx_hash, proposal_period);
```

Comment:
maybe do an estimate directly on transaction by calling estimateTransaction and not on trx_hash which makes you call getTransaction inside of estimateTransactionByHash to get the transaction